pkg/kubectl/cmd/get.go

```diff
@@ -220,13 +221,23 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	}
 
 	// use the default printer for each object
+	printer = nil
+	var lastMapping *meta.RESTMapping
+	w := kubectl.GetNewTabWriter(out)
+	defer w.Flush()
 	return b.Flatten().Do().Visit(func(r *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
-		printer, err := f.PrinterForMapping(cmd, r.Mapping, allNamespaces)
-		if err != nil {
-			return err
+		if printer == nil || lastMapping == nil || r.Mapping == nil || r.Mapping.Resource != lastMapping.Resource {
+			printer, err = f.PrinterForMapping(cmd, r.Mapping, allNamespaces)
+			if err != nil {
+				return err
+			}
+			lastMapping = r.Mapping
+		}
+		if _, found := printer.(*kubectl.HumanReadablePrinter); found {
+			return printer.PrintObj(r.Object, w)
```

Comment:
Users use `-o` or `--output` to specify different output format (use different printers).